### PR TITLE
Add information on `jobs` context for reusable workflows

### LINKED ISSUE #17823: `jobs` context isn't listed as a context in the Contexts article
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/contexts

### What part(s) of the article would you like to see updated?

Add a

> ## Jobs context
>
> Insert text explaining its structure. It's probably just a `jobs.`_job-id_`.`_job-context-content_ but maybe it isn't?
>
> There should be a reminder that the _job-id_ is just the label for the item under `jobs:` in the yaml -- it comes up periodically that people expect:
>
> ```
> jobs:
>   something:
>     id: an-id
> ```

### Additional information

Workflow key | Context | Special functions
-- | -- | --
concurrency | github, inputs |  
jobs.<job_id>.steps.working-directory | github, needs, strategy, matrix, **job**, runner, env, secrets, steps, inputs | hashFiles
on.workflow_call.outputs.<output_id>.value | github, _jobs_ [this is not defined as a context on this page], inputs

### PR DESCRIPTION
**Pull request by lucascosti**
### Why:

Closes https://github.com/github/docs/issues/17823

### What's being changed (if available, include any code snippets, screenshots, or gifs):

As per https://github.com/github/docs/issues/17823#issuecomment-1219399311, this adds information about the `jobs` context for reusable workflows.

https://docs-19979-6b8652.preview.ghdocs.com/en/actions/learn-github-actions/contexts#jobs-context

cc @ajaykn for a  👀  to make sure I'm covering things correctly here 🙂 

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by ajaykn — DISMISSED** [review]
Looks great to me ✨

**Review by hubwriter — DISMISSED** [review]
Just the raw/endraw thing. Otherwise this LGTM. 👍 
🚀

File: content/actions/learn-github-actions/contexts.md [hunk 1]
```diff
@@ -404,6 +406,70 @@ jobs:
...
+    }
+  }
+}
+```
+
+### Example usage of the `jobs` context
+
+This example reusable workflow uses the `jobs` context to set outputs for the reusable workflow. Note how the outputs flow up from the steps, to the job, then to the `workflow_call` trigger. For more information, see "[Reusing workflows](/actions/using-workflows/reusing-workflows#using-outputs-from-a-reusable-workflow)."
+
+```yaml{:copy}
```
[the remark this answers is not in the document]
Comment (lucascosti) [reply content/actions/learn-github-actions/contexts.md:443]: Great catch 🙇‍♂️

**Review by martin389 — APPROVED** [review]

File: content/actions/learn-github-actions/contexts.md [hunk 2]
```diff
@@ -404,6 +406,72 @@ jobs:
...
+### Example contents of the `jobs` context
+
+This example `jobs` context contains the result and outputs of a job from a reusable workflow run.
+
+```json
+{
+  example_job: {
+    result: success,
+    outputs: {
+      output1: hello,
```
Comment (jsoref) [inline content/actions/learn-github-actions/contexts.md:431]: At least some of these should be quoted, quite possible everything.

Comment (lucascosti) [reply content/actions/learn-github-actions/contexts.md:431]: PR to fix here: https://github.com/github/docs/pull/20362

**Pull request merged**